src/QuadForm/Herm/Lattices.jl

```diff
@@ -16,136 +16,103 @@ end
 #
 ################################################################################
 
+function lattice(V::HermSpace, B::PMat)
+  E = base_ring(V)
+  @req rank(matrix(B)) == min(nrows(B), ncols(B)) "B must be of full rank"
+  @req nf(base_ring(B)) == E "Incompatible arguments: B must be defined over E"
+  @req ncols(B) == dim(V) "Incompatible arguments: the number of columns of B must be equal to the dimension of V" 
+  gram = gram_matrix(V, matrix(B))
```

Comment:
This is too expensive to be called during the construction. 
Also I seem to remember that L.gram stays undefined if we have an ambient space.

Reply:
Yes I was not sure about this point, it was there on the previous codes so I kept it in case